### DISCUSSION TITLE
Debian Wheezy support in YARP/ICUB software

### DISCUSSION THREAD

**Discussion by traversaro** [robotology/community#82] [Q&A] [answered]
According to the documentation on the OS supported by YARP/ICUB software [1] the Debian Wheezy distribution should not be supported anymore. 

Anyway there are several iCub that still mount a Debian Wheezy as the operating system for the PC104 / support cluster . Is this a temporary situation (an update is planned) or a long term one? 

I am asking mostly to make sure that the CoDyCo/Koroibot software remains compatible with all the OS supported by YARP/ICUB (at the moment for example we are not testing in CI our software for compatibility with CMake 2.8.11, and some bugs already emerged for this reason https://github.com/robotology/idyntree/issues/111 ). 

@drdanz @lornat75 

[1] : http://wiki.icub.org/wiki/YARP_Supported_Distributions

**Comment by Tobias-Fischer** [comment]
A related question: Our `iCubLondon01` is also still using Wheezy. Is it safe to update to Jessie as written in the official documentation (https://www.debian.org/releases/stable/i386/release-notes/ch-upgrading.en.html)? I didn't find any PC104 customized ISO images of Jessie ..

Best, Tobi

**Comment by gsaponaro** [comment]
On `iCubLisboa01` we are also running "Debian GNU/Linux 7.2 (wheezy) + iCub customizations". When we eventually upgrade, I wonder if we should go via Debian upgrade on the existing OS, or rather install & customize a new iCub pc104 image.

**Comment by pattacini** [comment]
@mbrunettini might be also interested.

**Comment by barbalberto** [comment]
For pc104 using the standard OS upgrade is not possible.

**Comment by Tobias-Fischer** [comment]
Hmmm .. this is a big issue then, seeing that Wheezy is not officially supported anymore, and yet there is no way to upgrade PC104 to a newer version. As a matter of fact, updating the `icub-common` package on PC104 isn't possible right now, as it complains about some `QT` dependencies which cannot be resolved.

**Comment by plinioMoreno** [comment]
Because of our robot Vizzy (which also has a cfw002) I also asked 
@mbrunettini about newer kernel versions (Debian 8/Ubuntu 14.04), but he 
told me that the cfw002 kernel modules are not available yet for recent 
kernel versions.

On 12/14/2015 12:12 PM, Tobias Fischer wrote:

> Hmmm .. this is a big issue then, seeing that Wheezy is not officially 
> supported anymore, and yet there is no way to upgrade PC104 to a newer 
> version. As a matter of fact, updating the |icub-common| package on 
> PC104 isn't possible right now, as it complains about some |QT| 
> dependencies which cannot be resolved.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/robotology/QA/issues/82#issuecomment-164421925.

## 

---

Plinio Moreno López
http://www.isr.ist.utl.pt/~plinio
plinio@isr.tecnico.ulisboa.pt

---

**Comment by lornat75** [comment, marked as answer]
We have to keep supporting new debians for the pc104. I think there are problems building debian live images, but we will investigate them. 

The kernel module for the cfw002 should be compilable on new kernels, or at least I am not aware of problems (please let me know otherwise).

In the meanwhile you can disable qt on pc104 I think qt guis are not needed there.

**Comment by pattacini** [comment]
Can we close this issue?

**Comment by Tobias-Fischer** [comment]
I think there should be an issue somewhere around regarding the compatibility of PC104 and Debian Jessie :+1:

**Comment by pattacini** [comment]
I've made a quick search without success.

**Comment by gsaponaro** [comment]
https://github.com/robotology/icub-support/issues/119 ?

**Comment by pattacini** [comment]
Thanks @gsaponaro ! I actually forgot to check that repo too.
Closing then.
